sdk/src/rest/commands/utils/versions.ts

```diff
@@ -0,0 +1,68 @@
+import type { DirectusVersion } from '../../../schema/version.js';
+import type { UnpackList } from '../../../types/index.js';
+import type { RestCommand } from '../../types.js';
+
+/**
+ * Compare an existing version with the main item.
+ *
+ * @param id Primary key of the version.
+ *
+ * @returns All fields with different values, along with the hash of the main item and the information whether the current
+version is outdated (main item has been updated since the creation of the current version)
+ */
+export const compareContentVersion =
+	<Schema extends object, Collection extends keyof Schema, Item = UnpackList<Schema[Collection]>>(
+		id: DirectusVersion<Schema>['id']
+	): RestCommand<
+		{
+			outdated: boolean;
+			mainHash: string;
+			current: Partial<Item>;
+			main: Item;
+		},
+		Schema
+	> =>
+	() => ({
+		path: `/versions/${id}/compare`,
+		method: 'GET',
+	});
```

Comment:
I think we may want to check for empty `id` here too so we don't get `/versions//compare`
```suggestion
	> =>
	() => {
		throwIfEmpty(key, 'Key cannot be empty');
		
	    return {
    		path: `/versions/${id}/compare`,
	    	method: 'GET',
	    }
	};
```

Reply:
Great catch ❤️